app/code/core/Mage/Adminhtml/controllers/Customer/AttributeController.php

```diff
@@ -0,0 +1,328 @@
+            }
+
+            $defaultValueField = $model->getDefaultValueByInput($data['frontend_input']);
+            if ($defaultValueField) {
+                $data['default_value'] = $this->getRequest()->getParam($defaultValueField);
+            }
+
+            if(!isset($data['apply_to'])) {
+                $data['apply_to'] = array();
+            }
+
+            if ($model) {
+                $data['entity_type_id'] = $model->getEntityTypeId();
+            }
+
+            //filter
+            $data = $this->_filterPostData($data);
+            $model->addData($data);
+
+            if (!$id) {
+                $model->setEntityTypeId($this->_entityTypeId);
+                $model->setIsUserDefined(1);
+            }
+
+            try {
+                $model->save();
+                $session->addSuccess($helper->__('The product attribute has been saved.'));
+
+                /**
+                 * Clear translation cache because attribute labels are stored in translation
+                 */
+                Mage::app()->cleanCache(array(Mage_Core_Model_Translate::CACHE_TAG));
+                $session->setAttributeData(false);
+                if ($this->getRequest()->getParam('popup')) {
+                    $this->_redirect('adminhtml/customer/addAttribute', array(
+                        'id'       => $this->getRequest()->getParam('customer'),
+                        'attribute'=> $model->getId(),
+                        '_current' => true
+                    ));
+                } elseif ($redirectBack) {
+                    $this->_redirect('*/*/edit', array('attribute_id' => $model->getId(),'_current'=>true));
+                } else {
+                    $this->_redirect('*/*/', array());
+                }
+                return;
+            } catch (Exception $e) {
+                $session->addError($e->getMessage());
+                $session->setAttributeData($data);
+                $this->_redirect('*/*/edit', array('attribute_id' => $id, '_current' => true));
+                return;
+            }
+        }
+        $this->_redirect('*/*/');
+    }
+
+    public function deleteAction()
+    {
+        if ($id = $this->getRequest()->getParam('attribute_id')) {
+            $model = Mage::getModel('customer/attribute')
+                ->setEntityTypeId($this->_entityTypeId);
+
+            // entity type check
+            $model->load($id);
+            if ($model->getEntityTypeId() != $this->_entityTypeId || !$model->getIsUserDefined()) {
+                Mage::getSingleton('adminhtml/session')->addError(
+                    Mage::helper('customer')->__('This attribute cannot be deleted.'));
+                $this->_redirect('*/*/');
+                return;
+            }
+
+            try {
+                $model->delete();
+                Mage::getSingleton('adminhtml/session')->addSuccess(
+                    Mage::helper('customer')->__('The product attribute has been deleted.'));
```

Comment:
This phrase should say customer. It's used in multiple places.

Reply:
fixed it thanks!!